hydro_lang/src/live_collections/stream/mod.rs

```diff
@@ -1079,25 +1079,15 @@ where
         K: Ord,
         F: Fn(&T) -> K + 'a,
     {
-        let f = key.splice_fn1_borrow_ctx(&self.location);
-
-        let wrapped: syn::Expr = parse_quote!({
-            let key_fn = #f;
+        let f: ManualExpr<F, _> = ManualExpr::new(move |ctx: &L| key.splice_fn1_borrow_ctx(ctx));
+        self.reduce_commutative_idempotent(q!({
```

Comment:
Err, actually, this isn't commutative if there are two different inputs with the same key. Need to think a bit more, draft for now.